### ISSUE TITLE
There are some tips in how to decrease the loss in my training?

### ISSUE THREAD

**Issue by ndisanti-orangeloops** [apple/turicreate#3068]
Hello!

Not sure if this is the place to post the question, if i'm wrong let me know the correct place.

I'm in the process of finding the best model that i can train to detect a single object inside an image with a dataset of 950 tagged pictures ( mostly ~640x480) 

The best result that i reached, was a loss of 0.27, accuracy 0.99,  with 30k epochs , batch_size 48 (~16hrs on a MBP  2017 GPU), default grid_shape, default split between split/train

This training was the best try after training a lot different combinations of iterations/batch sizes/grid shapes

What are the things that i can try/tweak to improve this number? Looking for some help as i'm not an expert in the topic 

Thanks!

**Comment by TobyRoseman** [comment]
Our Object Detector doesn't currently expose many hyper parameters. Taking a look at the [API docs](https://apple.github.io/turicreate/docs/api/generated/turicreate.object_detector.create.html#turicreate.object_detector.create), it seem like `grid_shape`, `max_iterations` and `batch_size` are about all we have. It sounds like you've already tried different combinations for these values. So I don't really have any specific advice here. If you can get more training data, that's likely to improve model performance.

If the object you are trying to detect is a two-dimensional object with some regularity in the wild (ex: a road sign, logo, or playing cards), you might try using the [One-Shot Object Detection Toolkit](https://apple.github.io/turicreate/docs/userguide/one_shot_object_detection/). This toolkit exposes more option for data augmentation.

**Comment by TobyRoseman** [comment]
ndisanti-orangeloops - How many classes/labels are in your dataset? What operating system are you using?

**Comment by ndisanti-orangeloops** [comment]
Hi Toby 1 class only 
I never tried the One-Shot thing , based on your recommendation of the regularity of the object, i think is not an option as the object i tried to detect is not so regular in shape

My final results are similar to the ones i wrote at the beginning , a 0.24 loss with grid shape 10 and batch size 40

At the end, The model was detecting the object very well (failing in shadowed images, probably bc of the lack of training with this kind of data) , but was not achieving the detection speed we needed,  and running it inside an AVCaptureSession (predicting each 640x480 image delivered by the delegate) the phone's temperature after a minute was similar to the sun 

Finally we decided to take another approach that took us more time to integrate , but have similar detection rate but was way less demanding in the phone 

Thanks for your help with this, closing the issue
